client/components/close-on-escape/test/index.jsx

```diff
@@ -0,0 +1,75 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+import { shallow } from 'enzyme';
+import React from 'react';
+import ReactDom, { unmountComponentAtNode } from 'react-dom';
+import { spy } from 'sinon';
+
+/**
+ * Internal dependencies
+ */
+import useFakeDom from 'test/helpers/use-fake-dom';
+import CloseOnEscape from '../';
+
+const simulateEscapeKeydown = () => (
```

Comment:
I'm almost certain that events on `document`, `window` or anything other than a component is outside of enzyme's realm... 
I did a decent amount of googling in hope of finding a way to use enzymes `(component).simulate` or similar methods but couldn't find any solutions for this sort of behaviour.

The discussion on this issue (https://github.com/airbnb/enzyme/issues/426#issuecomment-225912455 ), particularly from that comment onwards was probably the most helpful.

It did have me thinking that there may be a better method for handling these events on a component vs globally with `document` / `window` but I've not come up with any solid ideas yet. The keypress itself would need to be somewhat 'global' as I doubt we could guarantee focus on some handler component.